### ISSUE TITLE
Certain Extensions not working 

### ISSUE THREAD

**Issue by digitaldrreamer** [friendofsvelte/tipex#22]
Hey there. 

Thanks for building such a stable project. Seriously, this is the only WYSIWYG editor I've found that works seamlessly with sveltekit (and is not chinese🥲). I especially love the `editor` store ❤️

I noticed that `extensions` accepts an object (and not an array of extensions) with some select property names (like heading etc). I tried using some extensions like tiptap search and replace (and gave a property name `searchAndReplace`, but it doesn't work. 

Is there something I'm missing? I really need tiptap extensions to work.

Thanks in advance

**Comment by digitaldrreamer** [comment]
Hi @Bishwas-py 😶

**Comment by Bishwas-py** [comment]
I have to update docs, but other than than it's fine. Can you share me you code snippet?

**Comment by digitaldrreamer** [comment]
Sorry for the late response @Bishwas-py 
I was talking about this example in the docs:
```svelte
<script lang="ts">
    import {defaultExtensions, Tipex} from "@friendofsvelte/tipex";
     let newExtensions = {
        ...defaultExtensions,
        heading: Heading.configure({
            levels: [2, 3, 4, 5, 6]
        })
    } // sample of tweaking extensions, so that only heading 2 to 6 are available
</script>
<Tipex extensions={newExtensions} .../>
```

The issue:
- that the `extensions` prop in the code accepts `any` type, and the docs state the prop accepts `Extension[]`. But the tipex editor accepts an object with `key` `extension` pairs (as is `defaultExtensions`. I was initially confused, thinking that there might be predefined keys to be used for each extensions (in the prop object), or that the extensions that could be used (in the prop object) were limited to the ones in the `defaultExtensions` and the example. Well, that was until I checked the code and saw this:`...Object.values(extensions) as Extensions`.

I was wondering that wouldn't it make more sense for the prop to just accept an array, instead of an object? Or to state explicitly in the doc that the keys do not matter, as long as the values are extensions.

![Screenshot_2024_10_20-1](https://github.com/user-attachments/assets/ef9b0533-d0c8-4168-ad6d-a7fcf9f6ccd6)
![Screenshot_2024_10_20-2](https://github.com/user-attachments/assets/22311788-046b-4920-923f-081f8d565c64)

**Comment by Bishwas-py** [comment]
@digitaldrreamer I agree with that! I will move on with that task after a while. Kinda busy these days.

(Contri are welcomed)

**Comment by Bishwas-py** [comment]
Well, it is because we have a conditional floating menu:
![image](https://github.com/user-attachments/assets/87d90a1e-9444-4b50-acd1-e6a5a0b9b2b3)

and we wanna detect that menu explicitly, any suggestions on that?

**Comment by Bishwas-py** [comment]
@digitaldrreamer You concern is now understood and implemented, in version `0.0.7-up-1`

```ts
import { Tipex } from "@friendofsvelte/tipex";
import { TextAlign } from '@tiptap/extension-text-align';

const extensions = [
    TextAlign.configure({
        types: ['heading', 'paragraph'],
    }),
];

// Use in component
<Tipex {extensions} />
```
